app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/model/InputValidationError.kt

```diff
@@ -0,0 +1,26 @@
+package com.davidtomas.taskyapp.features.auth.domain.model
+
+import com.davidtomas.taskyapp.core.domain.model.Failure
+
+sealed interface InputValidationError : Failure {
+    sealed interface EmailValidatorError : InputValidationError {
+        data object MissingMandatoryEmail : EmailValidatorError
+        data object EmailFormatError : EmailValidatorError
+    }
+
+    sealed interface PasswordValidatorError : InputValidationError {
+        data object MissingMandatoryPassword : PasswordValidatorError
+        data object PasswordFormatError : PasswordValidatorError
+        data object PasswordLengthError : PasswordValidatorError
```

Comment:
If your parent class already contains the type of the sub-classes (here that they're all password errors), you don't need to include that in the sub-classes. So `PasswordValidationError.PasswordFormatError` is duplicated information. Calling it `PasswordValidationError.Format` communicates exactly the same with less words